### DISCUSSION TITLE
Deep linking / Electron feedback

### DISCUSSION THREAD

**Discussion by mikehearn** [hydraulic-software/conveyor#36] [General]
@Christilut Let's resume the discussion here. Thanks for the feedback!

Re: URL handler schemes. That's very high on the list of things to add, because yes it's a common need. Here's an issue that we'll update when it's done:

https://github.com/hydraulic-software/conveyor/issues/5

Re: pricing. The price is for the combination of the product + support, not just support. Conveyor is partly open source (bits can be found in this repo) but mostly isn't, so you're buying a commercial product. Does that affect your perception of the pricing?

**Comment by Christilut** [comment]
Honestly, it would be an acceptable price but it does raise the expectation of it. So far I'm using electron-forge with a bunch of custom code because of weird issues I'm running into, all build related. Every weird issue took me a day or more to figure out and fix so a build tool that 100% works (and provides other really nice features like a moving away from squirrel) with support that I can rely on, is worth it to me. 
So if those expectations are in line with what you guys offer, and I don't need to compromise on what my app can do then that will definitely be interesting for me.

**Reply by mikehearn** [reply]
Great. Well, obviously we aim to do that. The product is new so we thought we'd get it out there and get feedback whilst adding other features. If you have other requirements beyond general ease of use, I'd love to hear them. 

The main ones we know about are:

* URL handlers and file associations, as mentioned.
* An API to control the update system (exists but not yet exposed/documented), e.g. to force an update at startup so clients are never out of date.
* RPM support.
* Uploading to the app stores.

The question is what's most important.

**Reply by Christilut** [reply]
For me the first 2 are most important. My clients are always on the latest version. Although right now it is an update prompt that you can delay. 

My app doesn't support Linux and it's not on app stores so the last 2 don't apply to me.

Not really sure about other requirements, although I'm sure I'll run into some things when I try to implement this.

**Reply by mikehearn** [reply]
@Christilut We've added some docs on how to do URL handlers with the current Conveyor release:

https://conveyor.hydraulic.dev/4.0/configs/os-integration/#url-handlers

The config required isn't elegant but we'll clean that up in future releases. We tested it on Win/Mac/Linux with a simple Compose Desktop app and it works as expected. For Electron there's no real difference, the packaging is the same, but we'll try with an Electron app soon anyway. You still need to use the single-instance code that the Electron tutorial shows.

**Reply by Christilut** [reply]
Cool, I'll give it a try sometime and probably email you with weird errors I'm getting :) Not sure when yet, always busy!

**Reply by mikehearn** [reply]
@Christilut FYI at the start of December we released Conveyor 5 which has two features you should find useful:

1. You can now specify URL handlers with much simpler syntax.
2. You can enable "aggressive updates" which provides a web-like update experience, where the app checks for and applies updates automatically on every app start.

**Reply by Christilut** [reply]
Thanks for the heads up. Still want to give it a try but no time yet!
